File: frontend/src/app/model/application/supervision/supervision-task-type.ts
export enum SupervisionTaskType {
  /**
   * Aloitusvalvonta
   */
  PRELIMINARY_SUPERVISION = 'PRELIMINARY_SUPERVISION',
  /**
   * Talvityön toiminnallinen kunto
   */
  OPERATIONAL_CONDITION = 'OPERATIONAL_CONDITION',
  /**
   * Valvonta
   */
  SUPERVISION = 'SUPERVISION',
  /**
   * Työnaikainen valvonta
   */
  WORK_TIME_SUPERVISION = 'WORK_TIME_SUPERVISION',
  /**
   * Loppuvalvonta
   */
  FINAL_SUPERVISION = 'FINAL_SUPERVISION',
  /**
   * Takuuvalvonta
   */
  WARRANTY = 'WARRANTY'
}

export function isAutomaticSupervisionTaskType(type: SupervisionTaskType): boolean {
  return [
    SupervisionTaskType.OPERATIONAL_CONDITION,
    SupervisionTaskType.WARRANTY,
    SupervisionTaskType.WORK_TIME_SUPERVISION
  ].indexOf(type) >= 0;
}

File: backend/common-domain/src/main/java/fi/hel/allu/common/domain/types/SupervisionTaskType.java
package fi.hel.allu.common.domain.types;

public enum SupervisionTaskType {
  /**
   * Aloitusvalvonta
   */
  PRELIMINARY_SUPERVISION(true),
  /**
   * Toiminnallisen kunnon valvonta
   */
  OPERATIONAL_CONDITION(false),
  /**
   * Valvonta
   */
  SUPERVISION(true),
  /**
   * Työnaikainen valvonta
   */
  WORK_TIME_SUPERVISION(false),
  /**
   * Loppuvalvonta
   */
  FINAL_SUPERVISION(true),
  /**
   * Takuuvalvonta
   */
  WARRANTY(false);

  private final boolean manuallyAdded;

  private SupervisionTaskType(boolean manuallyAdded) {
    this.manuallyAdded = manuallyAdded;
  }

  public boolean isManuallyAdded()  {
    return manuallyAdded;
  }

}
